core/shared/src/main/scala/zio/Semaphore.scala

```diff
@@ -110,13 +125,35 @@ object Semaphore {
         def withPermitsScoped(n: Long)(implicit trace: Trace): ZIO[Scope, Nothing, Unit] =
           ZIO.acquireRelease(reserve(n))(_.release).flatMap(_.acquire)
 
+        override def tryWithPermits[R, E, A](n: Long)(zio: ZIO[R, E, A])(implicit trace: Trace): ZIO[R, E, Option[A]] =
+          ZIO.acquireReleaseWith(tryReserve(n)) {
+            case Some(reservation) => reservation.release
+            case _              => Exit.none
```

Comment:
I think this can be 'Any', since that's what release returns. The `none` was probably from my suggestion previously.
```suggestion
            case _              => Exit.unit
```